clients/java/signalr/src/main/java/HubConnection.java

```diff
@@ -81,36 +87,50 @@ public HubConnection(String url, Transport transport) {
         }
     }
 
+    public HubConnection(String url, Transport transport) {
+        this(url, transport, new NullLogger());
+    }
+
     public HubConnection(String url) {
-        this(url, null);
+        this(url, null, new NullLogger());
+    }
+
+    public HubConnection(String url, LogLevel logLevel){
+        this(url, null, new ConsoleLogger(logLevel));
     }
 
     public HubConnectionState getConnectionState() {
         return connectionState;
     }
 
     public void start() throws Exception {
+        logger.log(LogLevel.Debug, "Starting connection");
         transport.setOnReceive(this.callback);
         transport.start();
         String handshake = HandshakeProtocol.createHandshakeRequestMessage(new HandshakeRequestMessage(protocol.getName(), protocol.getVersion()));
         transport.send(handshake);
         connectionState = HubConnectionState.CONNECTED;
+        logger.log(LogLevel.Information, "Connected started");
     }
 
     public void stop(){
+        logger.log(LogLevel.Debug, "Stopping connection");
         transport.stop();
         connectionState = HubConnectionState.DISCONNECTED;
+        logger.log(LogLevel.Information, "HubConnection stopped");
     }
 
     public void send(String method, Object... args) throws Exception {
         InvocationMessage invocationMessage = new InvocationMessage(method, args);
         String message = protocol.writeMessage(invocationMessage);
+        logger.log(LogLevel.Debug, "Sending message");
         transport.send(message);
     }
 
     public Subscription on(String target, Action callback) {
         ActionBase action = args -> callback.invoke();
         handlers.put(target, action);
+        logger.log(LogLevel.Trace, "Handler added to target: " + target);
```

Comment:
Can you go through [HubConnection.Log.cs](https://github.com/aspnet/SignalR/blob/release/2.2/src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.Log.cs) and make sure that you have similar messages here when they're relevant and that the text matches (unless it would be super weird).

We should make the logging consistent between those clients. We also should've done the same with TypeScript but better late than never ;P

Reply:
Yeah, these should be consistent